diff --git a/contrib/mbedtls/meson.build b/contrib/mbedtls/meson.build
index 06929f90b88dc71a7e3a23463b2a33bf9a989b25..81cf010b9674e309a33ef169b6a16f22612d2651 100644
--- a/contrib/mbedtls/meson.build
+++ b/contrib/mbedtls/meson.build
@@ -3,7 +3,7 @@
 # SPDX-License-Identifier: BSD-2-Clause
 
 if not builtin_mbedtls
-    mbedcrypto_lib = dependency('mbedcrypto', required: false)
+    mbedcrypto_lib = dependency('MbedTLS', required: false, method: 'cmake', modules: ['MbedTLS::mbedcrypto'])
     if not mbedcrypto_lib.found()
         mbedcrypto_lib = cc.find_library('mbedcrypto', required: required_library, has_headers: ['mbedtls/aes.h'])
         if not mbedcrypto_lib.found()
